packages/jest-reporters/src/notify_impl.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+export default function notify(
+  notification: Notification,
+  callback?: NotificationCallback,
+) {
+  try {
+    return require('node-notifier').notify(notification, callback);
+  } catch (ex) {
+    if (ex.code !== 'MODULE_NOT_FOUND') {
+      throw ex;
+    } else {
+      throw Error(
+        'notify reporter requires optional dependeny node-notifier but it was not found',
+      );
+    }
+  }
+}
+
+export type Notify = (
```

Comment:
copying the types seems suboptimal. Can we import them instead? as long as they're only used as type defs, babel will strip out the import from the resulint js file

Reply:
I chose to define the types instead of importing them from @types/node-notifier because I wanted  a subset of node-notifiers functionality. The idea being that jest could offer alternate notification implementations, and these types would be the common interface.

Still want me to change it?

Reply:
Yeah, I think so. This is just used internally, it's not part of any public API, so the abstraction is just extra code